configdialog.cpp

```diff
@@ -1256,9 +1268,66 @@ LRESULT CALLBACK CustomColorDialog(HWND dialog, UINT message, WPARAM wparam, LPA
 		}
 		case WM_COMMAND: {
 			switch (LOWORD(wparam)) {
+				case COLOR_RED:
+				case COLOR_GREEN:
+				case COLOR_BLUE: {
+					switch (HIWORD(wparam)) {
+						case EN_CHANGE: {
+							if (ignore_hex_change > 0) {
+								ignore_hex_change--;
+								return FALSE;
+							}
+							ignore_rgb_change = 1;
+
+							HWND red_input = GetDlgItem(dialog, COLOR_RED);
+							HWND green_input = GetDlgItem(dialog, COLOR_GREEN);
+							HWND blue_input = GetDlgItem(dialog, COLOR_BLUE);
+
+							std::wstring red = string_from_buffer<wchar_t, 4>([&](wchar_t *buffer, size_t size) {
+								Edit_GetText(
+									red_input,
+									buffer,
+									cast<int>(size)
+								);
+							});
+							std::wstring green = string_from_buffer<wchar_t, 4>([&](wchar_t *buffer, size_t size) {
+								Edit_GetText(
+									green_input,
+									buffer,
+									cast<int>(size)
+								);
+							});
+							std::wstring blue = string_from_buffer<wchar_t, 4>([&](wchar_t *buffer, size_t size) {
+								Edit_GetText(
+									blue_input,
+									buffer,
+									cast<int>(size)
+								);
+							});
+
+							if (red.empty() || green.empty() || blue.empty()) {
+								return FALSE;
+							}
+
+							std::wstring hex_code = std::format(L"#{:02x}{:02x}{:02x}", std::stoi(red), std::stoi(green), std::stoi(blue));
+							HWND hex_input = GetDlgItem(dialog, IDC_COLORDLG_HEX_CODE);
+							Edit_SetText(
```

Comment:
Despite the madness that is... the above code, so far, this has worked through everything I've thrown at it.

Reply:
I don't... disagree.